### PR TITLE
[SDK]Changed column mapping `${run.inputs.xx}`'s behavior, it will refer to run's data columns instead of run's inputs columns.

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

This PR changed column mapping `${run.inputs.xx}`'s behavior, it will refer to run's data columns instead of run's inputs columns to align with cloud's behavior.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: docs/reference/run-yaml-schema-reference.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ The source JSON schema can be found at [Run.schema.json](https://azuremlschemas.
...
 | `run`                   | string        | Referenced flow run name. For example, you can run an evaluation flow against an existing run.                                                                                                                                                                          |
-| `column_mapping`        | object        | Inputs column mapping, use ${data.xx} to refer to data file columns, use ${run.inputs.xx} and ${run.outputs.xx} to refer to run inputs/outputs columns.                                                                                                                 |
+| `column_mapping`        | object        | Inputs column mapping, use `${data.xx}` to refer to data file columns, use `${run.inputs.xx}` to refer to referenced run's data file columns, and `${run.outputs.xx}` to refer to run outputs columns.columns.                                                                                                                 |
```
Comment (elliotzh) [inline docs/reference/run-yaml-schema-reference.md:21]: misspell?

File: examples/tutorials/run-management/cloud-run-management.ipynb [hunk 2]
```diff
@@ -243,7 +243,7 @@
    "source": [
     "## 3 Create a flow run which uses an existing run's inputs data\n",
     "\n",
-    "When running a flow with an existing run, you can reference either it's inputs or outputs in column mapping.\n",
+    "When running a flow with an existing run, you can reference either it's inputs data or outputs in column mapping.\n",
```
Comment (elliotzh) [inline examples/tutorials/run-management/cloud-run-management.ipynb:246]: input data?

File: examples/tutorials/run-management/run-management.ipynb [hunk 3]
```diff
@@ -144,7 +144,7 @@
    "source": [
     "## 3 Create a flow run which uses an existing run's inputs data\n",
     "\n",
-    "When running a flow with an existing run, you can reference either it's inputs or outputs in column mapping.\n",
+    "When running a flow with an existing run, you can reference either it's inputs data or outputs in column mapping.\n",
```
Comment (elliotzh) [inline examples/tutorials/run-management/run-management.ipynb:147]: input data

**Review by wangchao1230 — DISMISSED** [review]

File: docs/reference/run-yaml-schema-reference.md (hunk 1 above)
Comment (D-W-) [reply docs/reference/run-yaml-schema-reference.md:21]: Good catch!

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py [hunk 4]
```diff
@@ -150,25 +152,22 @@ def test_run_bulk_error(self, pf):
...
 
     def test_basic_evaluation(self, azure_open_ai_connection: AzureOpenAIConnection, local_client, pf):
-        data_path = f"{DATAS_DIR}/webClassification3.jsonl"
-
         result = pf.run(
-            flow=f"{FLOWS_DIR}/web_classification",
-            data=data_path,
-            column_mapping={"url": "${data.url}"},
+            flow=f"{FLOWS_DIR}/print_env_var",
+            data=f"{DATAS_DIR}/env_var_names.jsonl",
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py:157]: why removed `column_mapping` parameter?

File: src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py [hunk 5]
```diff
@@ -150,25 +152,22 @@ def test_run_bulk_error(self, pf):
...
             column_mapping={
-                "groundtruth": "${data.answer}",
-                "prediction": "${run.outputs.category}",
+                "prediction": "${run.outputs.output}",
                 # evaluation reference run.inputs
                 # NOTE: we need this value to guard behavior when a run reference another run's inputs
-                "variant_id": "${run.inputs.url}",
+                "variant_id": "${run.inputs.key}",
+                # can reference other columns in data which doesn't exist in base run's inputs
+                "groundtruth": "${run.inputs.extra_key}",
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py:170]: why do we change this behavior in the first place? And it looks weird as the usage in the mapping is still `run.inputs.xx` while it's actually not run inputs but run data.

Comment (wangchao1230) [reply src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py:170]: I think we can still keep it. As both inputs and outputs are data actually.

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py (hunk 4 above)
Comment (D-W-) [reply src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py:157]: I changed the flow to improve CI speed(this flow doesn't have connections), and for this flow and data, it doesn't need column mapping.

File: src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py (hunk 5 above)
Comment (D-W-) [reply src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py:170]: Agree, if we want to change it, UI also need to change

**Review by PeiwenGaoMS — APPROVED** [review]

**Pull request merged**